tests/neg/i20474.scala

```diff
@@ -0,0 +1,13 @@
+class A
+class B extends A
+
+def f(a: A, c: A) =
+  val b1: a.type = a
+  val b2: a.type & B = a.asInstanceOf[a.type & B]
+  val b3: c.type & A = c
+  val b4: a.type | c.type = c
```

Comment:
I don't think that will work. Have you tried it? I see no string reason to make it work.

Reply:
It works in 3.4.2: https://scastie.scala-lang.org/A6PehvQvRhW2sUPVVtHiuA 
I think the `atoms` is doing all these checks.

Reply:
Ah, yes, atoms would do it. Bit it's an extreme corner case. Not sure we want to make sure it works in the general case. Because in this case we'd also have to accept

```scala
val b6: (a.type & A) | a.type
val d6: b6.type = a // ok
```
and I am not sure atoms would help us here.